hudi-common/src/main/java/org/apache/hudi/common/engine/RecordContext.java

```diff
@@ -244,6 +245,20 @@ public Comparable convertOrderingValueToEngineType(Comparable value) {
 
   public abstract GenericRecord convertToAvroRecord(T record, Schema schema);
 
+  /**
+   * Converts the engine-specific record to an Avro GenericRecord using HoodieSchema.
+   * This method uses HoodieSchema for in-memory processing while maintaining
+   * compatibility with existing Avro-based serialization.
+   *
+   * @param record The engine-specific record.
+   * @param schema The HoodieSchema of the record.
+   * @return An Avro GenericRecord.
+   * @since 1.2.0
+   */
+  public GenericRecord convertToAvroRecord(T record, HoodieSchema schema) {
```

Comment:
Similarly here, let's just migrate the method to use HoodieSchema instead of having two options